crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -288,4 +364,36 @@ mod test {
         }
         buffer
     }
+
+    #[test]
+    fn test_read_json_from_file() -> Result<(), anyhow::Error> {
+        let static_json = r#"{
+          "c8y_RequiredAvailability": {
+              "responseInterval": 1
+          },
+          "c8y_Firmware": {
+              "name": "test",
+              "version": "5.13.8-051308-generic",
+              "url": "31aab9856861b1a587e2094690c2f6e272712cb1"
+          },
+          "c8y_Hardware": {
+              "model": "E14",
+              "revision": "000e",
+              "serialNumber": "00000000e2f5ad4d"
+          }
+        }"#;
+
+        let dir = tempdir()?;
+
+        let file_path = dir.path().join("test.json");
+        let mut file = File::create(file_path)?;
+        file.write_all(static_json.as_bytes())?;
+
+        let expected_json: serde_json::Value = serde_json::from_str(&static_json)?;
+
+        let actual_json = read_json_from_file(dir.path().join("test.json").to_str().unwrap())?;
+
+        assert_eq!(actual_json, expected_json);
+        Ok(())
+    }
```

Comment:
I'm not sure if there is a value in this test, it tests file read and json conversion which are coming from `std` and `serde_json` crate.

I would remove it.